### ISSUE TITLE
Add dropout support to LoRA layers for training mode

### ISSUE THREAD

**Issue by GarthDB** [GarthDB/metal-candle#28]
LoRA layers should support configurable dropout during training for better regularization, matching the original LoRA paper implementation.

**Current Code:** `src/training/lora.rs:307` - Currently skips dropout

**Implementation Plan:**
1. Add training/inference mode flag to `LoRALayer`
2. Implement dropout using Candle's dropout operation
3. Apply dropout to hidden state between A and B matrix multiplications
4. Add tests for dropout behavior
5. Update documentation with dropout parameter

**Configuration:**
```rust
pub struct LoRAConfig {
    pub rank: usize,
    pub alpha: f32,
    pub dropout: f32,  // Already exists, just not applied
}
```

**Priority:** Low-Medium (v1.1)

**References:**
- Original LoRA paper uses dropout for regularization
- Standard practice in adapter training

**Comment by GarthDB** [comment]
✅ **Implemented and merged!**

Dropout support is now fully functional in metal-candle v1.0.

## Implementation Summary

**Features Added**:
- ✅ Training/eval mode control (`set_training()`, `eval()`, `is_training()`)
- ✅ Dropout applied between A and B matrix multiplications (per LoRA paper)
- ✅ Automatic disable in eval mode for deterministic inference
- ✅ Compatible with custom Metal kernel when dropout=0

**Code Changes**:
- Added `training: bool` field to `LoRALayer`
- Dropout uses Candle's `candle_nn::ops::dropout()`
- Preserves gradient flow for backpropagation
- Default mode: training (dropout enabled if configured)

**Testing**:
- 8 comprehensive tests in `tests/lora_dropout.rs`
- All 187 tests passing (179 lib + 8 new)
- Zero clippy warnings maintained

**Documentation**:
- Complete API documentation with examples
- Performance notes (dropout adds ~10-20% overhead)

## Usage Example

```rust
use metal_candle::training::{LoRALayer, LoRAConfig};
use candle_core::Device;

let device = Device::Cpu;
let config = LoRAConfig {
    rank: 8,
    alpha: 16.0,
    dropout: 0.1,  // Now applied!
};

let mut layer = LoRALayer::new(64, 64, &config, &device)?;

// Training mode (dropout enabled)
layer.set_training(true);
let train_output = layer.forward(&input)?;

// Eval mode (dropout disabled)
layer.eval();
let eval_output = layer.forward(&input)?;
```

## Ready for Ferris Integration

This implementation is production-ready and meets all requirements for Ferris migration. The dropout behavior matches the LoRA paper recommendations and is critical for training quality.

**Commit**: 4fdbeb9
**PR**: #26
